back-nest/src/getAllElement/products.services.ts

```diff
@@ -34,4 +35,29 @@ export class ProductService {
     this.logger.log('Returning products from database');
     return products;
   }
+
+  async upsertProducts(products: ScrapedProduct[]): Promise<void> {
+    await Promise.all(
+      products.map((productData) => {
+        const { title, source } = productData;
+
+        delete productData.newPrice;
+        const { price, newPrice, ...productToUpdate } = productData;
```

Comment:
You should either use delete or destructuring with removing properties. I recommend you to remove delete to object avoid mutation